DatadogWebViewTracking/Sources/WebViewTracking.swift

```diff
@@ -0,0 +1,56 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import DatadogInternal
+
+/// Defines methods to send WebView related information
+public protocol WebViewTracking {
+    /// Sends a message
+    /// - Parameter message: The message to send
+    func send(message: WebViewTrackingMessage) throws
+}
+
+/// Datadog implementation of `WebViewTracking`
+/// - Note:
+/// Cross platform SDKs should instantiate this type with a `DatadogCoreProtocol` implementation
+/// and pass WebView related messages using the message bus of the core.
+public struct WebViewTrackingCore: WebViewTracking {
+    enum MessageKeys {
+        static let browserLog = "browser-log"
+        static let browserRUMEvent = "browser-rum-event"
+    }
+
+    private weak var core: DatadogCoreProtocol?
+
+    public init(core: DatadogCoreProtocol) {
+        self.core = core
+    }
+
+    /// Sends a message to the message bus
+    /// - Parameter message: The message to send
+    public func send(message: WebViewTrackingMessage) throws {
+        assert(core != nil, "Core must not be nil when using WebViewTracking")
```

Comment:
We avoid assertions in production code (in case someone enables them for release build). Instead, we use `DD.logger` to make misconfigurations verbose (if `Datadog.verbosityLevel` is enabled).

Reply:
interesting, have we had such cases before?

happy to change it to a log message.

Reply:
I'm curious about this decision, I expect someway of breaking the code when there is developer error. Losing this ability and using logs seems like a back step.

Reply:
> interesting, have we had such cases before?

The very first one we added is perhaps [this](https://github.com/DataDog/dd-sdk-ios/blob/69ff9fe644098c3bc55d67af82e4600f8009bca7/Sources/Datadog/Datadog.swift#L83-L100) around `Datadog.initialize()` sequence. To not make it silent (and because of unavailability of `DD.logger` at this code point), we use `print()` directly.

> I'm curious about this decision, I expect someway of breaking the code when there is developer error. Losing this ability and using logs seems like a back step.

Early we decided on [zero crashes in our code](https://github.com/DataDog/dd-sdk-ios/blob/develop/ZEN.md#stability) and we stick to this idea to this day. Because we're "just observability" SDK, we prefer to fail silently rather than risk crashing the app in production. Also, as we rely on crash telemetry, any assertion failure of this kind would be considered a false-positive for our own monitoring.